src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -147,28 +148,27 @@ initialise1 as = initialise (head as) True <> initialise1 (tail as)
 
 -- Gate-level concepts
 buffer :: a -> a -> CircuitConcept a
-buffer a b = rise a ~> rise b <> fall a ~> fall b
+buffer a z = rise a ~> rise z <> fall a ~> fall z
 
 inverter :: a -> a -> CircuitConcept a
-inverter a b = rise a ~> fall b <> fall a ~> rise b
+inverter a z = rise a ~> fall z <> fall a ~> rise z
 
 cElement :: a -> a -> a -> CircuitConcept a
-cElement a b c = buffer a c <> buffer b c
+cElement a b z = buffer a z <> buffer b z
 
-meElement :: a -> a -> a -> a -> CircuitConcept a
-meElement r1 r2 g1 g2 = buffer r1 g1 <> buffer r2 g2 <> me g1 g2
-
-andGate :: a -> a -> a -> CircuitConcept a
-andGate a b c = rise a ~> rise c <> rise b ~> rise c
-             <> [fall a, fall b] ~|~> fall c
+mutexElement :: a -> a -> a -> a -> CircuitConcept a
```

Comment:
I think 'ME element' is a more common name for this gate, let's keep it. The change of `me` to `mutex` is fine though.